typescript/ExecuteFunction/index.test.ts

```diff
@@ -0,0 +1,231 @@
+      { baseURL: 'http://localhost:7071', headers: { 'Content-Type': 'application/json' } },
+    );
+    mockedAxios.mockRestore();
+  });
+
+  it('Returns a gzipped response body', async () => {
+    const mockedAxios = jest.spyOn(axios, 'post');
+    mockedAxios.mockResolvedValueOnce(mockProfileSuccessResponse).mockResolvedValueOnce(mockFunctionResponse);
+
+    const response = await bootstrappedFunction();
+    expect(Buffer.isBuffer(response.res.body)).toBe(true);
+    ungzip(response.res.body)
+      .then((uncompressed) => {
+        return gzip(uncompressed);
+      })
+      .then((compressed) => {
+        expect(compressed).toEqual(response.res.body);
+      });
+
+    mockedAxios.mockRestore();
+  });
+
+  it('Respondes with a full "ExecuteFunctionResult"', async () => {
+    const mockedAxios = jest.spyOn(axios, 'post');
+    mockedAxios.mockResolvedValueOnce(mockProfileSuccessResponse).mockResolvedValueOnce(mockFunctionResponse);
+
+    const response = await bootstrappedFunction();
+    const unzippedResponseBody = await ungzip(response.res.body);
+    const responseString = unzippedResponseBody.toString();
+    const responseBody = JSON.parse(responseString);
+    expect(responseBody).toHaveProperty('data.ExecutionTimeMilliseconds');
+    expect(responseBody).toHaveProperty('data.FunctionName');
+    expect(responseBody).toHaveProperty('data.FunctionResult');
+    expect(responseBody).toHaveProperty('data.FunctionResultTooLarge');
+
+    mockedAxios.mockRestore();
+  });
+
+  it('Does not gzip the result if gzip header is not present', async () => {
+    const mockedAxios = jest.spyOn(axios, 'post');
+    mockedAxios.mockResolvedValueOnce(mockProfileSuccessResponse).mockResolvedValueOnce(mockFunctionResponse);
+
+    const response = await bootstrappedFunction({ headers: headersWithoutComrpression });
+    expect(Buffer.isBuffer(response.res.body)).toBe(true);
+    expect(async () => {
+      await ungzip(response.res.body);
+    })
+      .rejects // https://stackoverflow.com/a/61214808
+      .toThrowError('Error: incorrect header check');
+
+    mockedAxios.mockRestore();
+  });
+
+  it('Returns a 401 when no Caller Entity Token header is present', async () => {
+    const mockedAxios = jest.spyOn(axios, 'post');
+    mockedAxios.mockResolvedValueOnce(mockProfileUnauthorizedResponse).mockResolvedValueOnce(mockFunctionResponse);
+    const response = await bootstrappedFunction({ headers: headersWithoutEntityToken });
+    expect(response).toHaveProperty('res.status', 401);
+    mockedAxios.mockRestore();
+  });
+
+  it('Returns a 200 when Entity is present in the request body', async () => {

```

Comment:
I assume if no Entity is provided in the request body, we still want the call to succeed? Just calling this out here because I'm not 100% what circumstances lead to this happening, or what the expected behavior is.

Reply:
Yes, if no entity is provided in the body, the entity in the entity token is used. Passing an entity in the body allows a title to execute a function as a player, for example. But if the title just wanted to execute a function as itself, it can just pass its entity token as normal and not specify an entity in the request body. Most functions are called by players, but we have some cases where customers are calling using a title entity token (see https://github.com/PlayFab/pf-af-devfuncs/pull/9)